docs/reference/mapping/runtime.asciidoc

```diff
@@ -0,0 +1,518 @@
+[[runtime]]
+== Runtime fields
+Typically, you index data into {es} to promote faster search. However, indexing
+can be slow and requires more disk space, and you have to reindex your data to
+add fields to existing documents. With _runtime fields_, you can add
+fields to documents already indexed to {es} without reindexing your data.
+
+[discrete]
+[[runtime-benefits]]
+=== Benefits
+Because runtime fields aren't indexed, you can more quickly ingest raw data
+into the Elastic Stack and access it immediately. By dynamically evaluating
+runtime fields at search time, you can optimize disk space by choosing which
+fields to index. If you later decide that a runtime field is increasingly used
+for filtering or aggregation, you can add the field to `_source` to gain faster
+search speed.
```

Comment:
will this make user think that we do something magic with existing documents?

Reply:
I'm rewriting a good portion of this section and will remove that sentence.